src/ar_http_iface_middleware.erl

```diff
@@ -1,23 +1,25 @@
--module(ar_http_iface_handler).
--behaviour(cowboy_handler).
--export([init/2, read_complete_body/2]).
+-module(ar_http_iface_middleware).
+-behaviour(cowboy_middleware).
+-export([execute/2, read_complete_body/2]).
 -include("ar.hrl").
 -define(HANDLER_TIMEOUT, 55000).
 
 %%%===================================================================
 %%% Cowboy handler callbacks.
 %%%===================================================================
 
-init(InitialReq, State) ->
+execute(Req, #{ handler := ar_http_iface_handler }) ->
```

Comment:
We've discussed it might be helpful to add a comment about this part.

Reply:
Comment added in bad9fd6a9aa74e9487b139558b1e219b92389a07